generic.go

```diff
@@ -57,17 +67,50 @@ func NewGeneric(config *Config, logger *zap.Logger, queue workqueue.RateLimiting
 			return nil, errors.Errorf("controller for GVK %s should have registered an informer for that GVK", descr.Gvk)
 		}
 		inf.AddEventHandler(&GenericHandler{
-			Logger:       logger,
+			Logger:       gvkLogger,
 			WorkQueue:    queueGvk,
 			ZapNameField: descr.ZapNameField,
 		})
 		controllers[descr.Gvk] = iface
+
+		// Extra controller data
+		groupKind := descr.Gvk.GroupKind()
+		replacer := strings.NewReplacer(".", "_", "-", "_", "/", "_")
+		objectName := replacer.Replace(groupKind.String())
+
+		objectProcessCount := prometheus.NewCounter(
+			prometheus.CounterOpts{
+				Namespace: config.PrometheusNamespace,
+				Name:      fmt.Sprintf("processed_%s_count", objectName),
+				Help:      fmt.Sprintf("Cumulative number of %s processed", groupKind.String()),
```

Comment:
No need to explicitly call `String()`, `%s` will call it. Hm, you may need to pass a pointer to it so that it call call the method.